### PR TITLE
Bump concepts to top and add concepts to overview

### PR DESCRIPTION
**Pull request by Noah-Silvera**

### CODE REVIEW

File: docs/web/concepts-advanced-config.mdx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
```
Comment (ianschmitz) [inline docs/web/concepts-advanced-config.mdx:1]: Shall we rename this file to `concepts-advanced-app-config` for consistency with the `concepts-app-config` equivalent?

File: docs/workflow/overview.mdx [hunk 2]
```diff
@@ -15,7 +15,21 @@ Geocortex Workflow was designed to be cross-platform, and it can run in [Geocort
 
 <DesignerCallout />
 
-## Start with your Use Case
+## Start with the Concepts
+
+Geocortex Workflow is built on the concept of [**workflows**](concepts-key-concepts.mdx#workflow). Workflows are a collection of connected **activities** and **forms**, and are created using Geocortex Workflow Designer.
```
Comment (ianschmitz) [inline docs/workflow/overview.mdx:20]:
```suggestion
Geocortex Workflow is built on the concept of a [**workflow**](concepts-key-concepts.mdx#workflow). A workflow is a collection of connected **activities** and **forms**, that can be created using Geocortex Workflow Designer.
```

**Comment by Noah-Silvera** [comment]
@kenlyon Could you review the new concepts section in the workflow overview? It's probably easiest to review by checking out the preview

https://deploy-preview-24--geocortex-dev-center.netlify.app/docs/workflow/overview#start-with-the-concepts (The link for the preview can also be found by clicking "details" on the "deploy-preview" step of the github checks list)

**Comment by Noah-Silvera** [comment]
> It looks good to me. The only thought I have is whether we should make the distinction between "activities" and "steps" at all. Originally, the idea was that a step is an instance of an activity. For example, in the designer the toolbox contains a list of activities, whereas the designer canvas shows steps connected together.
> 
> In practice, nobody seems to use the term "step" except for developers who are working on the code where the distinction matters. This could be an opportunity to re-introduce the distinction if it was helpful.

That's an interesting thought....Correct me if I'm wrong, but the distinction starts to matter when you are working on the workflow engine which needs to connect "steps"? I think from a custom activity authors view, the word "activity" can encompass the code definition they make with the SDK and the instantiated code at runtime (the step), without being confusing, so I would say let's not introduce another concept? With the current ways you can extend the SDK, I just don't see the concepts of steps arising that much / being that useful of an disctinction? But that may change with documentation that adds more in depth descriptions of the activity framework. Tempted to leave it out for now, and add it if the situation arises?

**Comment by ianschmitz** [comment]
Agreed let's not introduce concepts unless necessary. `Activity` is probably enough?

**Comment by kenlyon** [comment]
> That's an interesting thought....Correct me if I'm wrong, but the distinction starts to matter when you are working on the workflow engine which needs to connect "steps"? I think from a custom activity authors view, the word "activity" can encompass the code definition they make with the SDK and the instantiated code at runtime (the step), without being confusing, so I would say let's not introduce another concept? With the current ways you can extend the SDK, I just don't see the concepts of steps arising that much / being that useful of an disctinction? But that may change with documentation that adds more in depth descriptions of the activity framework. Tempted to leave it out for now, and add it if the situation arises?

The distinction is also relevant when considering the authoring of a workflow. In the designer you might rearrange the steps in the diagram, for example. It sounds like you're wanting to make this very focused on authoring of activity packs, though, in which case "activity" will be adequate.

File: docs/web/concepts-advanced-config.mdx (hunk 1 above)
Comment (Noah-Silvera) [reply docs/web/concepts-advanced-config.mdx:1]: Totally. Done.

**Pull request merged**